### PR TITLE
Prework task 5 b done

### LINKED ISSUE #31: Prework - Task 5 b
Utwórz 3 pochodne klasy Cargo:

Fruit

Klasa Fruit powinna mieć dodatkową zmienną określającą czas do zepsucia oraz operator-- który będzie odejmował ten czas o 1. Metoda getPrice() powinna redukować cenę odpowiednio wraz z czasem psucia naszego owocu.

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

File: shm/src/Fruit.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "shm/inc/Fruit.hpp"
+
+#include <iostream>
+
+Fruit::Fruit(const std::string& name, size_t amount, size_t basePrice) 
+        : Cargo(name, amount, basePrice) {
+}
+
+Fruit& Fruit::operator--() {
+    if (rottenTime !=0){
```
Comment (MoreThanADC) [inline shm/src/Fruit.cpp]: Add one space between != and 0

File: shm/src/Fruit.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        rottenTime--;
+    } else {
+        std::cout << "Fruit is completely rotten\n";
+    }
+    return *this;
+}
+
+std::size_t Fruit::getPrice() {
+    return basePrice_ * (rottenTime / 10);
```
Comment (mariuszlisowski) [inline shm/src/Fruit.cpp]: Integer division - may lost precision. Tests needed to check the result.

File: shm/src/Fruit.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+Fruit::Fruit(const std::string& name, size_t amount, size_t basePrice) 
+        : Cargo(name, amount, basePrice) {
+}
+
+Fruit& Fruit::operator--() {
+    if (rottenTime !=0){
+        rottenTime--;
+    } else {
+        std::cout << "Fruit is completely rotten\n";
```
Comment (mariuszlisowski) [inline shm/src/Fruit.cpp]: std::cerr

File: shm/src/Fruit.cpp (hunk 2 above)
Comment (lukasz-kukulka) [reply shm/src/Fruit.cpp]: Do we need a good precision? If we use this formula everywhere, we shouldn't have any problems

File: shm/src/Fruit.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Fruit::Fruit(const std::string& name, size_t amount, size_t basePrice) 
+        : Cargo(name, amount, basePrice) {
+}
+
+Fruit& Fruit::operator--() {
+    if (rottenTime_ != 0){
+        rottenTime_--;
+    return *this;
+}
```
Comment (MoreThanADC) [inline shm/src/Fruit.cpp]: The function must return something. Maybe close if bracket after "rottenTime_--"

**Pull request closed without merging**